command/bitmarkd/commands.go

```diff
@@ -455,23 +461,54 @@ func getFilenameWithDirectory(arguments []string, name string) string {
 	return filepath.Join(dir, name)
 }
 
-func makeSigningKey(test bool, fileName string) error {
-	seedCore := make([]byte, 32)
-	if _, err := rand.Read(seedCore); err != nil {
-		return fmt.Errorf("failed to generate signing core error: %s", err)
-	}
-	seed := []byte{0x5a, 0xfe, 0x01, 0x00} // header + network(live)
-	if test {
-		seed[3] = 0x01 // change network to testing
+func makeSigningKey(testnet bool, fileName string) error {
+	seed, err := generateEncodedSeed(testnet)
+	if nil != err {
+		return err
 	}
-	seed = append(seed, seedCore...)
-	checksum := sha3.Sum256(seed)
-	seed = append(seed, checksum[:4]...)
 
-	data := "SEED:" + util.ToBase58(seed) + "\n"
-	if err := ioutil.WriteFile(fileName, []byte(data), 0600); err != nil {
+	data := "SEED:" + seed + "\n"
+	if err = ioutil.WriteFile(fileName, []byte(data), 0600); nil != err {
 		return fmt.Errorf("error writing signing key file error: %s", err)
 	}
 
 	return nil
 }
+
+func generateEncodedSeed(testnet bool) (string, error) {
```

Comment:
I wonder if this should be part of account to keep the seed ←→Ed25519 private key logic all in one place

Reply:
Yes. So i think `private.go` is better.